pkg/virt-api/rest/subresource.go

```diff
@@ -417,6 +420,66 @@ func (app *SubresourceAPIApp) StopVMRequestHandler(request *restful.Request, res
 	response.WriteHeader(http.StatusAccepted)
 }
 
+func (app *SubresourceAPIApp) SpiceRequestHandler(request *restful.Request, response *restful.Response) {
+	name := request.PathParameter("name")
+	namespace := request.PathParameter("namespace")
+
+	vmi, code, err := app.fetchVirtualMachineInstance(name, namespace)
+	if err != nil {
+		response.WriteError(code, err)
+		return
+	}
+
+	if !vmi.IsRunning() {
+		response.WriteError(http.StatusBadRequest, goerror.New(fmt.Sprintf("Unable to connect to VirtualMachineInstance because phase is %s instead of %s", vmi.Status.Phase, v1.Running)))
+		return
+	}
+
+	// should not occur if the vm is handled by the virt-handler and in running state
+	if vmi.Status.SpiceConnection == nil {
+		response.WriteError(http.StatusInternalServerError, goerror.New(fmt.Sprintf("spice connection is not configured")))
```

Comment:
This starts with lower-case, while the previous error starts with upper case, please keep it consistent. Also, if this block should never happen, maybe we should state it in the error message? "spice connection is not configured" doesn't sound that fatal.

Reply:
it's fatal because `should not occur if the vm is handled `